x/slashing/keeper/keeper.go

```diff
@@ -71,27 +79,13 @@ func (k Keeper) Logger(ctx context.Context) log.Logger {
 
 // AddPubkey sets a address-pubkey relation
 func (k Keeper) AddPubkey(ctx context.Context, pubkey cryptotypes.PubKey) error {
-	bz, err := k.cdc.MarshalInterface(pubkey)
-	if err != nil {
-		return err
-	}
-	store := k.storeService.OpenKVStore(ctx)
 	key := types.AddrPubkeyRelationKey(pubkey.Address())
```

Comment:
I'm not really happy about the fact that we still need to rely on `module/types` to convert a key into its bytes  storage representation, as collections was born specifically to not need to do this sort of things. 

This is due to the fact that we migrated addresses to []bytes, and collections provides only "legacy" bytes format (length prefixing) only for AccAddress, ConsAddress and ValAddress. 

Maybe we should add a legacy format key codec for `[]byte` in `cosmos-sdk/types/collections.go`.